app/post.go

```diff
@@ -393,6 +393,14 @@ func (a *App) FillInPostProps(post *model.Post, channel *model.Channel) *model.A
 		post.DelProp("channel_mentions")
 	}
 
+	if !a.HasPermissionToChannel(post.UserId, post.ChannelId, model.PERMISSION_USE_CHANNEL_MENTIONS) {
+		post.AddProp(model.POST_PROPS_MENTION_HIGHLIGHT_DISABLED, true)
+	}
```

Comment:
you actually dont need this here the prop is added in the create post function via `post.DisableMentionHighlights()`

Reply:
True that friend! Thanks for pointing it out. I didn't pay close attention to notice that.